teps/0128-scheduled-runs.md

```diff
@@ -0,0 +1,279 @@
+
+### Drawbacks
+
+- We might want to avoid setting a precedent of building functionality into Triggers to trigger on many different types
+of events. This proposal couples an event (a time occurring) with a Tekton resource creation, when we may prefer to keep
+events and resource creation separate. However, cron jobs are an extremely common simple use case that likely makes sense
+to address specifically.
+
+- We might want to build this functionality into a higher level API like [Workflows](./0098-workflows.md) or a larger
+feature like [polling runs](./0083-polling-runs-in-tekton.md). This proposal doesn't prevent that,
+and CronTriggers could be leveraged to implement similar functionality in a higher level system.
+
+- The use of the name "EventListener" for a CRD that doesn't need to "listen" for incoming events may be confusing.
+
+## Alternatives
+
+### Create a new CronTrigger CRD
+
+Example:
+
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: CronTrigger
+metadata:
+  name: nightly-release
+spec:
+  schedule: "0 0 * * *"
+  bindings:
+  - name: project-name
+    value: wait-task
+  - name: triggers-uuid
+    value: $(context.eventID)
+  - name: datetime
+    value: $(context.datetime)
+  template:
+    ref: experimental-release-pipeline-template
+  serviceAccountName: default
+  cloudEventSink: http://eventlistener.free.beeceptor.com
+```
+
+CronTriggers don't support interceptors, and cannot be used in EventListeners.
+The `schedule` field uses the same syntax as Kubernetes CronJobs.
+`serviceAccountName` defaults to "default", and `cloudEventSink` is optional.
+
+Bindings for CronTriggers are optional and don't support `$(body.foo)` or `$(header.foo)` syntax.
+They support variable replacement for $(context.eventID), a UUID for the event, which is already supported in Triggers.
+
+Pros:
+- No need to worry about handling validation errors for interceptors, especially in referenced Triggers
+
+Cons:
+- Awkward syntax for creating many Triggers with the same schedule, unlike TriggerGroups
+- Reuses many features from EventListener
+
+### Add schedule to Trigger CRD
+
+Example:
+
+```yaml
+kind: Trigger
+spec:
+  schedule: "0 0 * * *"
+  bindings:
+  - name: project-name
+    value: wait-task
+  - name: triggers-uuid
+    value: $(context.eventID)
+  - name: datetime
+    value: $(context.datetime)
+  template:
+    ref: experimental-release-pipeline-template
+```
+
+Triggers with both a `schedule` and `interceptors` would result in a validation error.
+Inline TriggerBindings using `$(body.foo)` and `$(header.foo)` syntax would also result in a validation error.
+
+Unlike a Trigger without a `schedule`, a Trigger with a `schedule` can create resources without being used in an
+EventListener, using the default serviceAccount. (We could also add a `serviceAccountName` field to the Trigger CRD,
+which would override a `serviceAccountName` specified in any EventListeners.) If used in an EventListener, an incoming
```

Comment:
This already exists: https://github.com/tektoncd/triggers/blob/main/pkg/apis/triggers/v1beta1/trigger_types.go#L42

Reply:
whoops, thanks!